source/Irrlicht/CMakeLists.txt

```diff
@@ -212,21 +216,34 @@ macro(irr_target_link_shaderc _trgt)
 	target_link_libraries(${_trgt} INTERFACE
 		shaderc
 	)
-	target_include_directories(${_trgt} PUBLIC ${IRR_ROOT_PATH}/3rdparty/shaderc/libshaderc/include)
+	target_include_directories(${_trgt} PUBLIC ${THIRD_PARTY_SOURCE_DIR}/shaderc/libshaderc/include)
 endmacro()
 macro(irr_target_link_libjpeg _trgt)
 	add_dependencies(${_trgt} jpeg)
 	target_link_libraries(${_trgt} INTERFACE
 		jpeg
 	)
-	target_include_directories(${_trgt} PUBLIC ${IRR_ROOT_PATH}/3rdparty/libjpeg)
+	target_include_directories(${_trgt} PUBLIC ${THIRD_PARTY_SOURCE_DIR}/libjpeg)
 endmacro()
 macro(irr_target_link_libpng _trgt)
 	add_dependencies(${_trgt} png_static)
 	target_link_libraries(${_trgt} INTERFACE
 		png_static
 	)
-	target_include_directories(${_trgt} PUBLIC ${IRR_ROOT_PATH}/3rdparty/libpng)
+	target_include_directories(${_trgt} PUBLIC ${THIRD_PARTY_SOURCE_DIR}/libpng)
+endmacro()
+macro(irr_target_link_openexr _trgt)
+	#add_dependencies(${_trgt} openexr)
```

Comment:
I wanted to add dependency as IlmBase or OpenEXR, that was my mistake, there is no need to add it since the only one dependency that could be added is `zlib`, but it is already done earlier

Reply:
they're adding their own zlib I guess.

you need to add dependency IlmBase, etc. just as jpeg, png, shaderc are adde above
then link libraries,
then add include directories

seems you'll need to add dependency and link IlmBase as well

Reply:
I've looked through your log, you're getting compile errors before you're getting linking errors, hence you don't know if a library called `openexr` even exists (wont get messages about library not found)

Also your log jumps straight from Documetnation target to IRrlicht, there are no Ilm or OpenEXR targets being built, this tells me that IRrlicht does not have the Ilm targets in the dependency chain.

*Before*
Introduce `_IRR_COMPILE_WITH_OPENEXR_LOADER_` and disable it.
See if you can compile with the loader disabled, if you can't you've fucked up an exisitng and working build system.

*First*
try to add dependencies explicitly, so you see the Ilm/OpenEXR targets being built or checked for building in the log

*Second*
Fix your missing include issues (include dirs), but make sure to sort out dependencies above first, as the build process may be producing include files in the build dir that you need to find.

*Third*
Maybe limit the OpenEXR includes to the .cpp file, so that the whole project doesnt get polluted by OpenEXR declarations.

Reply:
_status:_

- Before - **DONE properly, projects build**
- First - ...
- Second - ...
- Third - ...

Reply:
what the progress on the second?

Reply:
I did it all for you , see PR to your repo